impeller/renderer/pool.h

```diff
@@ -0,0 +1,33 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#pragma once
+
+#include <memory>
+
+namespace impeller {
+
+template <typename T>
+class Pool {
+ public:
+  std::shared_ptr<T> Grab() {
+    if (pool_.empty()) {
+      return T::Create();
+    } else {
+      std::shared_ptr<T> result = pool_.back();
+      pool_.pop_back();
+      return result;
+    }
+  }
+
+  void Recycle(std::shared_ptr<T> object) {
```

Comment:
Do we need to trim this set eventually? It may grow large on a frame with many render passes and then not shrink back down again.

Reply:
I was worried about this but in my local testing I got 2 buffers that were 262144 bytes long.  That didn't seem worth the effort to clean up.

Reply:
I think we should add some kind of limit even if its very simple. For example, if the length is 32 then throw away the oldest and replace it.

Reply:
Done (limited pool size in bytes)